src/modules/recipes/NewRecipeForm/index.jsx

```diff
@@ -0,0 +1,226 @@
+import { useForm, useFieldArray } from "react-hook-form";
+import { useRecipesDispatch } from "../RecipesProvider";
+import { RECIPE_ACTIONS } from "../RecipesProvider";
+import "./styles.css";
+
+export const NewRecipeForm = () => {
+  const dispatch = useRecipesDispatch();
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm({
+    defaultValues: {
+      strMeal: "",
+      strCategory: "Beef",
+      strArea: "Asian",
+      ingredients: [{ ingredient: "", measure: "", amount: "" }],
+      strInstructions: "",
+      strMealImage: "",
+    },
+  });
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: "ingredients",
+  });
+
+  const onSubmit = (data) => {
```

Comment:
Once the form is successfully submitted, make sure to clear the form and also show a success message on the screen so that the user knows what happened